src/Adapter/MSTest.TestAdapter/Helpers/MethodRunner.cs

```diff
@@ -16,9 +16,18 @@ internal static class MethodRunner
     internal static TestFailedException? RunWithTimeoutAndCancellation(
         Action action, CancellationTokenSource cancellationTokenSource, int? timeout, MethodInfo methodInfo,
         string methodCancelledMessageFormat, string methodTimedOutMessageFormat)
-        => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Thread.CurrentThread.GetApartmentState() == ApartmentState.STA
-            ? RunWithTimeoutAndCancellationWithSTAThread(action, cancellationTokenSource, timeout, methodInfo, methodCancelledMessageFormat, methodTimedOutMessageFormat)
-            : RunWithTimeoutAndCancellationWithThreadPool(action, cancellationTokenSource, timeout, methodInfo, methodCancelledMessageFormat, methodTimedOutMessageFormat);
+    {
+        if (timeout is null)
+        {
+            action();
```

Comment:
is it the cancellation verified before after somewhere out of this method?

Reply:
Same behavior as before introducing the timeout feature.